### PR TITLE
Update kubernetes-nmstate to v0.3.0

### PR DESCRIPTION
**Pull request by qinqon**
It contains a rewrite of it using operator-sdk

### CODE REVIEW

**Comment by qinqon** [comment]
\cc @phoracek

**Comment by phoracek** [comment]
@qinqon please create a separate PR bumping nmstate version. After it is merged, follow https://github.com/kubevirt/cluster-network-addons-operator#releasing to make a new release.

**Comment by qinqon** [comment]
@phoracek nmstate v0.3.0 is here https://github.com/nmstate/kubernetes-nmstate/pull/118

**Comment by phoracek** [comment]
I'm talking about this PR. We pin newer versions of deployed operand images in separate PRs. Once those PRs are tested and merged, we release a new version of the operator, using `hack/release.sh`. This PR should be split in two.

**Comment by qinqon** [comment]
@phoracek reworked this PR as "just" the upgrade to nmstate v0.3.0

File: data/nmstate/002-crd.yaml [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
-  version: v1
   group: nmstate.io
-  scope: Namespaced
   names:
     kind: NodeNetworkState
+    listKind: NodeNetworkStateList
     plural: nodenetworkstates
+    singular: nodenetworkstate
     shortNames:
     - net-state
```
[the remark this answers is not in the document]
Comment (qinqon) [reply data/nmstate/002-crd.yaml]: Well this is not part of this PR, but I can address it here.

**Comment by qinqon** [comment]
@phoracek manifest cleanup at kubernets-nmstate https://github.com/nmstate/kubernetes-nmstate/pull/126

File: data/nmstate/001-rbac.yaml [hunk 2]
```diff
@@ -2,65 +2,116 @@
...
-  name: nmstate-state-controller
+  name: nmstate-handler
   apiGroup: rbac.authorization.k8s.io
-subjects:
-  - kind: ServiceAccount
-    name: nmstate-state-controller
-    namespace: nmstate
 ---
-apiVersion: rbac.authorization.k8s.io/v1beta1
+---
```
[the remark this answers is not in the document]
Comment (qinqon) [reply data/nmstate/001-rbac.yaml]: Done

File: data/nmstate/001-rbac.yaml [hunk 3]
```diff
@@ -2,65 +2,116 @@
...
+  - ""
+  resources:
+  - pods
+  verbs:
+  - get
+  - list
+- apiGroups:
+  - ""
+  resources:
+  - pods/exec
```
[the remark this answers is not in the document]
Comment (qinqon) [reply data/nmstate/001-rbac.yaml]: Nope, already tested in the kubernetes-nmstate cleanup.

Done

File: data/nmstate/001-rbac.yaml [hunk 4]
```diff
@@ -2,65 +2,116 @@
...
+  - ""
+  resources:
+  - nodes
+  verbs:
+  - get
+  - list
+- apiGroups:
+  - ""
+  resources:
+  - pods
```
[the remark this answers is not in the document]
Comment (qinqon) [reply data/nmstate/001-rbac.yaml]: ditto

File: data/nmstate/002-crd.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (qinqon) [reply data/nmstate/002-crd.yaml]: Nah let's do it.

File: data/nmstate/003-operator.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      serviceAccountName: nmstate-handler
+      nodeSelector:
+        beta.kubernetes.io/arch: amd64
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          operator: Exists
+          effect: NoSchedule
+      containers:
+        - name: nmstate-handler
+          # Replace this with the built image name
```
[the remark this answers is not in the document]
Comment (qinqon) [reply data/nmstate/003-operator.yaml]: Done

File: data/nmstate/003-operator.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+          image:  {{ .NMStateHandlerImage }}
+          imagePullPolicy: {{ .ImagePullPolicy }}
+          command:
+          - kubernetes-nmstate
+          imagePullPolicy: Always
```
[the remark this answers is not in the document]
Comment (qinqon) [reply data/nmstate/003-operator.yaml]: done

File: data/nmstate/002-crd.yaml [hunk 7]
```diff
@@ -1,17 +1,40 @@
...
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
```
Comment (phoracek) [inline data/nmstate/002-crd.yaml]: this will need some changes, but let's not delay this PR with it

**Pull request merged**